nestkernel/connection_manager.cpp

```diff
@@ -94,9 +94,9 @@ nest::ConnectionManager::initialize()
   secondary_recv_buffer_pos_.resize( num_threads );
   sort_connections_by_source_ = true;
 
-  have_connections_changed_.resize( num_threads, true );
-  check_primary_connections_.resize( num_threads, false );
-  check_secondary_connections_.resize( num_threads, false );
+  have_connections_changed_.initialize( num_threads, true );
```

Comment:
Well, actually it would make a lot more sense to initialize this to `false`, too.
I'll check if this breaks anything...

Reply:
Travis indeed reported one failing Python test when `have_connections_changed_` is initialized to `false`. I did not see any problems when running `hpc_benchmark` with 32 MPI processes and 24 threads per process and none of the Python tests fails on my laptop. I guess it makes sense to look into this but I don't want to delay this PR.
@jakobj do you remember why `have_connections_changed_` needs to be initialized to `true`?

Reply:
@stinebuu @hakonsbm Could you contribute to clarifying this?

Reply:
Unfortunately I don't know why it is initialized to `true`.

Reply:
Let's not delay this PR because of this particular initialization.
I have issued another PR #1448 initializing `have_connections_changed_` to `false` such that we can continue the investigation there.

Reply:
:+1: